llms/mlx_lm/server.py

```diff
@@ -618,6 +618,38 @@ def handle_text_completions(self) -> mx.array:
         prompt = self.tokenizer.encode(prompt_text)
         return mx.array(prompt)
 
+    def do_GET(self):
+        """
+        Respond to a GET request from a client.
+        """
+        if self.path == "/v1/models":
+            self.handle_models_request()
+        else:
+            self._set_completion_headers(404)
+            self.end_headers()
+            self.wfile.write(b"Not Found")
+
+    def handle_models_request(self):
+        """
+        Handle a GET request for the /v1/models endpoint.
+        """
+        self._set_completion_headers(200)
+        self.end_headers()
+        # Create a list of available models
+        models = [
+            {
+                "id": self.model_provider.cli_args.model,
+                "object": "model",
+                "created": self.created,
+            }
```

Comment:
This looks like it just retrieves the active model. Would it be good to list all the already downloaded models or something like that?

Reply:
> It could check the huggingface/hub directory for mlx-community models.
 
We have some code in MLX LM already for [querying the HF cache](https://github.com/ml-explore/mlx-examples/blob/main/llms/mlx_lm/manage.py). We could repurpose some of that for this. Ideally it would list al models (not just MLX Community) in the HF cache that have a `model_type`  which MLX LM supports. That wouldn't include models not in the cache.. but I think that's ok.

> The bigger change would be on how the server should react when a v1/chat/completions call is made to a model that hasn't been pre-loaded.

The server already handles this.

Are you interested in adding that? In this PR or a follow on?